### PR TITLE
feat: [Redis] add support for data tiering nodes

### PR DESCRIPTION
**Pull request by zucchinidev**
[#184160811](https://www.pivotaltracker.com/story/show/184160811)

### Checklist:

* [ ] Have you added Release Notes in the docs repositories?
* [x] Have you followed the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)?

### CODE REVIEW

File: aws-redis-cluster.yml [hunk 1]
```diff
@@ -94,6 +94,16 @@ provision:
...
+  - field_name: data_tiering_enabled
+    type: boolean
+    details: |
+      Enables data tiering. Data tiering is only supported for replication groups using the r6gd node type.
+      This parameter must be set to true when using `r6gd` nodes.
+      Not all versions of Redis support this feature. Check supported versions.
+      For more information about Node Types and supported Redis versions see
+      https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/CacheNodes.SupportedTypes.html
+    default: false
+    prohibit_update: true
```
Comment (fnaranjo-vmw) [inline aws-redis-cluster.yml:106]: Prohibiting updating this field introduces unexpected behaviour:
- node_type field is undateable
- if node_type is set to an `r6gd` node_type but is later changed to a non `r6gd` node_type,
 data_tiering_enabled either (I don't know which will happen):
  - causes the node_type change to fail
  - node_type change succeeds but `data_tiering_enabled: true` no longer represents the reality

File: integration-tests/redis_test.go [hunk 2]
```diff
@@ -8,15 +8,19 @@ import (
...
+	redisServiceID                        = "e9c11b1b-0caa-45c9-b9b2-592939c9a5a6"
+	redisServiceName                      = "csb-aws-redis"
+	redisServiceDescription               = "Beta - CSB Amazon ElastiCache for Redis - multinode with automatic failover"
+	redisServiceDisplayName               = "CSB Amazon ElastiCache for Redis (Beta)"
+	redisServiceDocumentationURL          = "https://docs.vmware.com/en/Tanzu-Cloud-Service-Broker-for-AWS/1.2/csb-aws/GUID-reference-aws-redis.html"
+	redisServiceSupportURL                = "https://aws.amazon.com/redis/"
+	redisServiceProviderDisplayName       = "VMware"
+	redisCustomPlanName                   = "custom-sample"
+	redisCustomPlanID                     = "c7f64994-a1d9-4e1f-9491-9d8e56bbf146"
+	deprecatedCacheSizePlanName           = "deprecated-cachesize-sample"
```
Comment (fnaranjo-vmw) [inline integration-tests/redis_test.go:20]: nitpick: Maybe by introducing a new line between `redisCustomPlanID` and `deprecatedCacheSizePlanName` go formatter wouldn't try to reformat untouched lines to preserve alignment.

Having git/Github diffs only highlight added/changed lines is a big help when reviewing the code.
However, feel free to leave it as it is. But maybe keep it in mind for future commits.

Comment (fnaranjo-vmw) [reply integration-tests/redis_test.go:20]: Also, I see you have done some cleanup of the code from cache_size.
I would recommend adding these changes in a different PR to keep this PR focused on the changes needed to make data_tiering work.
Also, feel free to be more strict when reviewing my PRs.
I wouldn't be bothered by you asking me to modify my PR to use less hardcoded strings in tests.

Comment (fnaranjo-vmw) [reply integration-tests/redis_test.go:20]: I believe this PR is small enough that it is ok to leave the cleanup changes in it.

File: aws-redis-cluster.yml (hunk 1 above)
Comment (pivotal-marcela-campo) [reply aws-redis-cluster.yml:106]: @fnaranjo-vmw  The node_type change wouldn't succeed as you cannot move from a data_tiering enabled node type to a non-data-tiering node (and vice versa). So I think prohibiting the update in this field actually has the effect of providing rapid feedback to the user.

**Review by pivotal-marcela-campo — APPROVED** [review]

File: aws-redis-cluster.yml (hunk 1 above)
Comment (fnaranjo-vmw) [reply aws-redis-cluster.yml:106]: Perfect. Everything looks nice then!
Great job @zucchinidev !

**Review by fnaranjo-vmw — APPROVED** [review]

File: aws-redis-cluster.yml (hunk 1 above)
Comment (fnaranjo-vmw) [reply aws-redis-cluster.yml:106]: Hey @pivotal-marcela-campo , do you know if we could encode this logic (changing from data-tiering to non-data tiering) as a terraform tests?
Or this logic only happens in the IAAS, and as such can't be expressed as a terraform test?

File: integration-tests/redis_test.go (hunk 2 above)
Comment (blgm) [reply integration-tests/redis_test.go:20]: https://github.blog/2018-05-01-ignore-white-space-in-code-review/

**Pull request merged**